src/mds/MDCache.cc

```diff
@@ -8358,6 +8358,40 @@ int MDCache::path_traverse(const MDRequestRef& mdr, MDSContextFactory& cf,
   if (flags & MDS_TRAVERSE_CHECK_LOCKCACHE)
     mds->locker->find_and_attach_lock_cache(mdr, cur);
 
+  if (mdr) {
+    const cref_t<MClientRequest> &req = mdr->client_request;
+    Session *session = mds->get_session(req);
+    if (session && !session->info.has_feature(CEPHFS_FEATURE_ALTERNATE_NAME) &&
+	req->get_op() != CEPH_MDS_OP_LOOKUP &&
+	req->get_op() != CEPH_MDS_OP_LOOKUPHASH &&
+	req->get_op() != CEPH_MDS_OP_LOOKUPPARENT &&
+	req->get_op() != CEPH_MDS_OP_LOOKUPINO &&
+	req->get_op() != CEPH_MDS_OP_LOOKUPNAME &&
+	req->get_op() != CEPH_MDS_OP_LOOKUPSNAP &&
+	req->get_op() != CEPH_MDS_OP_RMSNAP &&
+	req->get_op() != CEPH_MDS_OP_LSSNAP &&
+	req->get_op() != CEPH_MDS_OP_GETATTR &&
+	req->get_op() != CEPH_MDS_OP_READDIR &&
+	req->get_op() != CEPH_MDS_OP_UNLINK &&
+	req->get_op() != CEPH_MDS_OP_RMDIR) {
+      MutationImpl::LockOpVec lov;
+      /* We need 'As' caps for the fscrypt context */
+      lov.add_rdlock(&in->authlock);
```

Comment:
```
/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos9/DIST/centos9/MACHINE_SIZE/gigantic/release/19.0.0-5143-gc0713fae/rpm/el9/BUILD/ceph-19.0.0-5143-gc0713fae/src/mds/MDCache.cc:8383:23: error: ‘in’ was not declared in this scope; did you mean ‘yn’?
```

Should be `lov.add_rdlock(&cur->authlock)`

Reply:
Done.